tools/clang/lib/Parse/ParsePragma.cpp

```diff
@@ -251,8 +251,10 @@ void Parser::initializePragmaHandlers() {
   } // HLSL Change, matching HLSL check to remove pragma processing
   else {
     // HLSL Change Begin - packmatrix.
-    PackMatrixHandler.reset(new PragmaPackMatrixHandler(Actions));
-    PP.AddPragmaHandler(PackMatrixHandler.get());
+    // The pointer ownership goes to PP, which deletes it in its destructor
+    // unless it is removed & deleted via resetPragmaHandlers
+    pPackMatrixHandler = new PragmaPackMatrixHandler(Actions);
+    PP.AddPragmaHandler(pPackMatrixHandler); 
```

Comment:
This could still leak in the one case where the collection fails to expand when adding the handler.  We probably aren't hitting that, but might if we added a bunch of pragmas this way and ran the OOM test against it.  To prevent that you would need to use a local unique_ptr, then release it after adding the handler.  Also, if you look at AddPragmaHandler the allocation of the namespace (we aren't currently using) can leak the namespace under the same scenario.